fabric-levelgenerators-v1/src/main/java/net/fabricmc/fabric/impl/levelgenerator/FabricLevelGeneratorType.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.levelgenerator;
+
+import java.util.HashMap;
+import java.util.function.Function;
+
+import net.minecraft.util.Pair;
+import net.minecraft.world.World;
+import net.minecraft.world.biome.source.BiomeSource;
+import net.minecraft.world.gen.chunk.ChunkGenerator;
+import net.minecraft.world.gen.chunk.ChunkGeneratorType;
+import net.minecraft.world.level.LevelGeneratorType;
+
+import net.fabricmc.fabric.mixin.levelgenerator.AccessorLevelGeneratorType;
+
+public class FabricLevelGeneratorType {
+	public static final HashMap<LevelGeneratorType, Pair<ChunkGeneratorType<?,
+					? extends ChunkGenerator<?>>, Function<World, BiomeSource>>> suppliers = new HashMap<>();
+
+	public static LevelGeneratorType create(String name, ChunkGeneratorType<?,
+					? extends ChunkGenerator<?>> generatorType, Function<World, BiomeSource> biomeSourceSupplier) {
+		return create(name, name, 0, generatorType, biomeSourceSupplier);
+	}
+
+	public static LevelGeneratorType create(String name, int version, ChunkGeneratorType<?,
+					? extends ChunkGenerator<?>> generatorType, Function<World, BiomeSource> biomeSourceSupplier) {
+		return create(name, name, version, generatorType, biomeSourceSupplier);
+	}
+
+	public static LevelGeneratorType create(String name, String storedName, int version, ChunkGeneratorType<?,
```

Comment:
Use an Identifier for the name

Reply:
Vanilla isn't using it and LevelGeneratorTypes are not registered.
Should we enforce using level-type=modid:name in server.prop then?
TranslationKey for client without changes to getTranslationKey() will look like this "generator.modid:name".

Reply:
yeah, I would.

For the translation keys, I would almost replace the `:` with `.`